File: Data Structures & Algorithms/validate-parentheses/submission-5.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char>st;
        for(int i=0;i<s.size();i++)
        {
            char ch=s[i];
            if(s[i]=='(' || s[i]=='[' || s[i]=='{')
            {
                st.push(s[i]);
            }
            else
            {
                if(st.empty())
                {
                    return false;
                }
                else if((ch==')'&& st.top()=='(')||(ch==']'&& st.top()=='[')||(ch=='}'&& st.top()=='{'))
                {
                    st.pop();
                }
                else{
                    return false;
                }
            }
        }
        return(st.empty());
    }
};

File: Data Structures & Algorithms/asteroid-collision/submission-5.java
class Solution {
    public int[] asteroidCollision(int[] asteroids) {
        Stack<Integer>st=new Stack<>();
        for(int i:asteroids)
        {
            boolean destroy=false;
            while(!st.empty() && i<0 && st.peek()>0)
            {
                if(Math.abs(st.peek())<Math.abs(i))
                {
                    st.pop();
                    continue;
                }
                else if(Math.abs(st.peek())==Math.abs(i))
                {
                    st.pop();
                }
                destroy=true;
                break;
            }
            if(destroy==false)
            {
                st.push(i);
            }
        }
        int[] arr=new int[st.size()];
        for(int i=st.size()-1;i>=0;i--)
        {
            arr[i]=st.peek();
            st.pop();
        }
        return arr;
    }
}